File: BasketballAPI/BasketballAPI/Models/Game.cs
namespace BasketballAPI.Models
{
    public class Game
    {
        public int Id { get; set; }
        public int LocalPoints { get; set; }
        public int VisitorPoints { get; set; }
        public int LocalFouls { get; set; }
        public int VisitorFouls { get; set; }
        public int CurrentQuarter { get; set; }
        public int Minutes { get; set; }
        public int Seconds { get; set; }
    }
}

File: src/app/services/game.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Game {
  id?: number;
  localPoints: number;
  visitorPoints: number;
  localFouls: number;
  visitorFouls: number;
  currentQuarter: number;
  minutes: number;
  seconds: number;
}

@Injectable({
  providedIn: 'root'
})
export class GameService {

  private apiUrl = 'http://localhost:5231/game';

  constructor(private http: HttpClient) { }

  getGame(id: number): Observable<Game> {
    return this.http.get<Game>(`${this.apiUrl}/${id}`);
  }

  createGame(game: Game): Observable<Game> {
    return this.http.post<Game>(this.apiUrl, game);
  }

  updateGame(id: number, game: Game): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${id}`, game);
  }
}
